source/common/router/config_impl.cc

```diff
@@ -65,6 +67,82 @@ Optional<uint64_t> HashPolicyImpl::generateHash(const Http::HeaderMap& headers)
   return hash;
 }
 
+HeaderFormatterSharedPtr RequestHeaderParser::parseInternal(const std::string& format) {
+	std::string variable_name;
+	if (format.find("%") == 0) {
```

Comment:
It feels like we're duplicating a bunch of parsing code from AccessLogFormatParser::parse here for the icky parsing of the % delimited variables. Can we just do this in one place and reuse?